### PR TITLE
Add initial JPEG 2000 (jp2) support

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds initial JPEG 2000 (jp2) support
tfio.experimental.image.decode_jp2

At the moment only the following has been supported:
- jp2 only, no jpx yet
- channel = 3 or channel = 4
- 8 bit

Will expand to add more support in follow up PRs.

This PR is part of #1017

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow_io/core/kernels/image_jpeg2k_kernels.cc [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    opj_stream_set_seek_function(l_stream.get(), OpjStreamCallback::SeekFn);
+
+    opj_dparameters_t l_param;
+    opj_set_default_decoder_parameters(&l_param);
+
+    // do not use layer decoding limitations
+    // l_param.cp_layer = 0;
+
+    // do not use resolutions reductions
+    // l_param.cp_reduce = 0;
```
Comment (terrytangyuan) [inline tensorflow_io/core/kernels/image_jpeg2k_kernels.cc:147]: Remove commented code?

File: tensorflow_io/core/kernels/image_jpeg2k_kernels.cc [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                      "] = ", p_image->comps[i].prec));
+    }
+    switch (p_image->numcomps) {
+      case 3:
+      case 4:
+        break;
+      default:
+        OP_REQUIRES(context, false,
+                    errors::InvalidArgument(
+                        "only 3 or 4 channel images supported, received ",
```
Comment (terrytangyuan) [inline tensorflow_io/core/kernels/image_jpeg2k_kernels.cc:177]:
```suggestion
                        "only images with 3 or 4 channels are currently supported, received ",
```

Comment (yongtang) [reply tensorflow_io/core/kernels/image_jpeg2k_kernels.cc]: @terrytangyuan Updated.

File: tensorflow_io/core/kernels/image_jpeg2k_kernels.cc (hunk 1 above)
Comment (terrytangyuan) [reply tensorflow_io/core/kernels/image_jpeg2k_kernels.cc:147]: Still here?

Comment (yongtang) [reply tensorflow_io/core/kernels/image_jpeg2k_kernels.cc:147]: @terrytangyuan I added a TODO, as those two parameters might be needed. Haven't find a sample to test yet, so leave them in TODO comment.

**Pull request merged**